### PR TITLE
401 - adds a function to filter the allowable cells by cell-types

### PR DESCRIPTION
**Pull request by AxelleDRouge**
## 🧰 Issue
The available cells need to have travel restrictions (sea/land/air) [#401 ](https://github.com/serge-web/serge/issues/401)

## 🚀 Overview: 
I added a function to define the cell type, then use it to filter the list of cells allowable

## 🔗 Link to preview
[https://serge-review-feature-40-8kib4x.herokuapp.com/storybook/?path=/story/local-mapping--with-allowable-range](https://serge-review-feature-40-8kib4x.herokuapp.com/storybook/?path=/story/local-mapping--with-allowable-range)

## 🔨Work carried out:
- [x] Added a function to add the cell type to each cells in the grid
- [x] Added a variable to extract the cells from the available cells by type

- [x] Tests pass

## Confirmations

- [x] I have chosen reviewers for my PR.
- [x] I have assigned myself to this PR.
- [x] I have chosen an appropriate label for the PR.
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.
- [x] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

**Comment by foxleigh81** [comment]
@AxelleDRouge Hi Axelle, is this set to [WIP] as it's not ready to merge yet? If so then can you convert it into a draft PR instead?

Impressed with what I've seen so far :)

Edit: Not sure if you converted it or if it was already a draft and my coffee deprive brain just didn't see it. Either way, thanks :)

**Comment by AxelleDRouge** [comment]
@foxleigh81  It was already a draft PR

**Comment by foxleigh81** [comment]
> @foxleigh81 It was already a draft PR

Yup, definitely need more coffee then.

File: packages/components/src/local/mapping/index.stories.tsx [hunk 1]
```diff
@@ -53,25 +53,25 @@ const OSMTileLayer = {
  * DEFAULT VIEW
  */
 export const Default: React.FC = () => <Mapping
-  tileDiameterMins = {5}
```
Comment (IanMayo) [inline packages/components/src/local/mapping/index.stories.tsx:56]: @foxleigh81  - does @AxelleDRouge need a VS-Code settings file?  Or, should these formatting changes have been caught in the pre-push hook?

Comment (foxleigh81) [reply packages/components/src/local/mapping/index.stories.tsx:56]: Yeah, husky should have picked those up before it was pushed. @AxelleDRouge did you run `yarn install` in the repo root? If you only ran it inside the `components` package, you won't have everything you need.

**Comment by IanMayo** [comment]
On first impressions it looks like the wrong cells are being allowed/not-allowed.  This is because the cell-types data-file was produced for a grid of a different size.  

This video shows the cells _are_ being blocked, but they appear to be in the wrong place over the map:
screencast

But, the video shows a shortcoming: the user should not be able to draw the line/route onto the non-allowed cells. I expect this is due to an instance where we're checking against `allowableCells`, rather than `filteredAllowableCell`.

File: packages/components/src/local/mapping/index.stories.tsx (hunk 1 above)
Comment (AxelleDRouge) [reply packages/components/src/local/mapping/index.stories.tsx:56]: I think I did yes.
I'll try again to be sure

Comment (AxelleDRouge) [reply packages/components/src/local/mapping/index.stories.tsx:56]: It worked except for the export node command, but replaced by set for windows it works

**Comment by AxelleDRouge** [comment]
Indeed, I correct this problem

File: packages/components/src/local/mapping/helpers/cell-types.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { SergeHex } from '@serge/custom-types'
```
Comment (IanMayo) [inline packages/components/src/local/mapping/helpers/cell-types.ts]: This isn't a technical shortcoming, but having two files titled `cell-types.ts` feels scary.

Could we rename the helper function (& filename) to `cellTypeFor`?

Aah, I think I've also spotted a potential performance improvement.  I may be wrong, but it looks like we read/parse `cellTypesList` lots and lots of times.  But, it doesn't change.

Is it an option to load the data object once when `create-grid` runs, then pass the structure in to this function as a parameter?

Comment (AxelleDRouge) [reply packages/components/src/local/mapping/helpers/cell-types.ts]: Yes inded those could be improved, I was trying to be consistent with the cellName function but it does create that shortcoming

Comment (AxelleDRouge) [reply packages/components/src/local/mapping/helpers/cell-types.ts]: I changed the name and extracted the list in create-grid

**Review by IanMayo — APPROVED** [review]
Really impressive first piece of work, well done @AxelleDRouge

**Pull request merged**